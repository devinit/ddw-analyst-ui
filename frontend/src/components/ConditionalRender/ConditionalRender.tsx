import * as React from 'react';

export const ConditionalRender: React.SFC<{ render: boolean }> = props =>
  props.render ? <React.Fragment>{ props.children }</React.Fragment> : null;
