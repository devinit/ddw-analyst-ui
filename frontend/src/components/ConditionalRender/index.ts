import { ConditionalRender } from './ConditionalRender';

export default ConditionalRender;
export * from './ConditionalRender';
