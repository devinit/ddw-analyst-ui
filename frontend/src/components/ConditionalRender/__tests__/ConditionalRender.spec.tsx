import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { ConditionalRender } from '../ConditionalRender';

test('renders null when the condition evaluates to false', () => {
  const renderer = TestRenderer
    .create(<ConditionalRender render={ false }>My Child</ConditionalRender>)
    .toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders the children when the condition evaluates to true', () => {
  const renderer = TestRenderer
    .create(<ConditionalRender render={ true }>My Child</ConditionalRender>)
    .toJSON();

  expect(renderer).toMatchSnapshot();
});
