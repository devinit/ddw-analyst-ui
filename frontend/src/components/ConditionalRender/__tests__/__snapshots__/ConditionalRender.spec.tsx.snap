// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders null when the condition evaluates to false 1`] = `null`;

exports[`renders the children when the condition evaluates to true 1`] = `"My Child"`;
